python/lsst/ts/MTAOS/model.py

```diff
@@ -352,14 +433,182 @@ async def run_wep(self, visit_id, extra_id, config, **kwargs):
+            self.log.debug(
+                f"Processing intra/extra pair: {visit_id}/{extra_id}. Expecting ComCam data."
+            )
+
+            await self.process_comcam(
+                intra_id=visit_id, extra_id=extra_id, config=config
+            )
+
+    async def process_comcam(self, intra_id, extra_id, config):
+        """Process ComCam intra/extra focual images.
+
+        Parameters
+        ----------
+        intra_id : `int`
+            Id of the intra-focal image.
+        extra_id : `int`
+            Id of the extra-focal image.
+        config : `dict`
+
+        """
+        self.log.debug(f"Processing ComCam intra/extra pair: {intra_id}/{extra_id}.")
+
+        wep_configuration = self.generate_wep_configuration(
+            instrument="comcam", reference_id=intra_id, config=config
+        )
+
+        comcam_config_file = tempfile.NamedTemporaryFile(suffix=".yaml")
+
+        comcam_config_file.write(yaml.safe_dump(wep_configuration).encode())
+
+        comcam_config_file.flush()
+
+        self.log.debug(f"Pipeline configuration in {comcam_config_file.name}.")
+
+        run_pipetask_cmd = writePipetaskCmd(
+            self.data_path,
+            self.run_name,
+            self.pipeline_instrument["comcam"],
+            "refcats," + self.collections,
+            pipelineYaml=comcam_config_file.name,
+        )
 
-        # TODO: (DM-28710) Initial implementation of runWEP command in MTAOS
-        raise NotImplementedError("This function is not supported yet (DM-28710).")
+        run_pipetask_cmd += f' -d "exposure IN ({intra_id}, {extra_id})"'
+        run_pipetask_cmd += f" -j {self.pipeline_n_processes}"
+
+        self.log.debug(f"Running: {run_pipetask_cmd}")
+
+        # Run pipeline task in a process asynchronously
+        wep_process = await asyncio.create_subprocess_shell(
+            run_pipetask_cmd,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+
+        stdout, stderr = await wep_process.communicate()
+
+        self.log.debug(f"Process returned: {wep_process.returncode}")
+
+        self.log.debug(stdout)
+
+        self.log.error(stderr)
+
+        self.log.debug("Data processing completed. Gathering output.")
+
+        comcam_config_file.close()
+
+        butler = dafButler.Butler(self.data_path)
+
+        # We may need to run the following in an executor so we won't block the
+        # event loop.
+
+        datasetRefs = list(
+            butler.registry.queryDatasets(
+                datasetType="postISRCCD", collections=[self.run_name]
+            )
+        )
+        for ref in datasetRefs:
+            self.log.debug(ref.dataId)
+
+        # Get output
+        data_ids = butler.registry.queryDatasets(
+            self.zernike_table_name,
+            dataId=dict(
+                instrument=self.data_instrument_name["comcam"], exposure=extra_id
```

Comment:
I think this will default to use the lower of the two exposure ids which happens to be the `extra_id` in the test data. We should check if this is the case and if we need to think more about how to assign a `dataId` for this output we should talk.

Reply:
But maybe it's ok if it does that and we just need to know for sure that it does that?

Reply:
good point. I think now It changed to use the intra_id.